common/workunit/wujobq.cpp

```diff
@@ -1122,11 +1122,17 @@ class CJobQueue: public CJobQueueBase, implements IJobQueue
     void disconnect()   // signal no longer wil be dequeing (optional - done automatically on release)
     {
         Cconnlockblock block(this,true);
-        if (connected) {
+        if (connected)
+        {
             dounsubscribe();
-            ForEachQueue(qd) {
+            ForEachQueue(qd)
+            {
                 IPropertyTree *croot = queryClientRootSession(*qd);
-                croot->setPropInt64("@connected",croot->getPropInt64("@connected",0)-1);
+                unsigned newConnectedCount = croot->getPropInt64("@connected",0)-1;
```

Comment:
Could the clearStaleMetaData remove all items, and then a client call this code?

If so, newConnectedCount would be -1.

Reply:
that should never happen.
Dali is only clearing up when it loads, so there should implicitly no clients connected/no SDS connection.
But I've added a guard JIC